### PR TITLE
TST: Add numeric coercion tests

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] related to #12747 and #12780
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`

Because dtype coercion affects to lots of functions, would like to fix in the following steps. The behavior which looks incorrect are commented with `TODO_GH12747`

1(this PR) Add numeric related tests to clarify the current behaviour (and future change) 
2. Fix numeric coercion.
3. Add datetime-like tests.
4. Fix datetime-like coercion.

### CODE REVIEW

File: pandas/tests/series/test_indexing.py [hunk 1]
```diff
@@ -1820,3 +1820,350 @@ def test_multilevel_preserve_name(self):
         result2 = s.ix['foo']
         self.assertEqual(result.name, s.name)
         self.assertEqual(result2.name, s.name)
+
```
Comment (jreback) [inline pandas/tests/series/test_indexing.py]: why don't we put these in ``tests/indexing/test_coercion.py``?

File: pandas/tests/indexes/test_numeric.py [hunk 2]
```diff
@@ -886,3 +886,83 @@ def test_ufunc_coercions(self):
         tm.assertIsInstance(result, Float64Index)
         exp = Float64Index([0.5, 1., 1.5, 2., 2.5], name='x')
         tm.assert_index_equal(result, exp)
+
+
+class TestIndexCoercion(tm.TestCase):
+
+    _multiprocess_can_split_ = True
```
Comment (jreback) [inline pandas/tests/indexes/test_numeric.py]: and these too (as these are mixed Series/index things), IOW not particular to the index tests, but more so indexing

**Comment by sinhrks** [comment]
@jreback OK, created `test_coercion.py`.

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**